bundles/org.openhab.binding.wundergroundupdatereceiver/README.md

```diff
@@ -0,0 +1,180 @@
+| AqPM10            | pm10-mass                    | Number:Density       | PM10 Mass                      | PM10 mass, µG/m3.                                                                                   | Pollution   | 
+| AqOZONE           | ozone                        | Number:Dimensionless | Ozone                          | Ozone, ppb.                                                                                         | Pollution   | 
+
+#### Metadata channel-types:
+| Request parameter |  Channel type id             | Type                 | Label                             | Description                                                                                         | Group       |
+|-------------------|------------------------------|----------------------|-----------------------------------|-----------------------------------------------------------------------------------------------------|-------------|
+| dateutc           | datetime-utc                 | String               | Last Updated                      | The date and time of the last update in UTC as submitted by the weather station. This can be 'now'. | Metadata    | 
+| softwaretype      | softwaretype                 | String               | Software Type                     | A software type string from the weather station                                                     | Metadata    | 
+| rtfreq            | realtime-frequency           | Number               | Realtime Frequency                | How often does the weather station submit measurements                                              | Metadata    | 
+| lowbatt           |
+
+#### Programatically added channel-types:
+|  Channel type id       | Type                 | Channel type | Label          | Description                                                 | Group    |
+|------------------------|----------------------|--------------|----------------|-------------------------------------------------------------|----------|
+| last-received-datetime | DateTime             | state        | Last Received  | The date and time of the last update.                       | Metadata | 
+| last-query-state       | String               | state        | The last query | The part of the last query after the first unurlencoded '?' | Metadata |
+| last-query-trigger     | String               | trigger      | The last query | The part of the last query after the first unurlencoded '?' | Metadata |
+
+The trigger channel's payload is the last querystring, so the following dsl rule script
+would send the measurements on to wunderground.com:
+
+```
+val requestQuery = receivedEvent
+sendHttpGetRequest("https://rtupdate.wunderground.com/weatherstation/updateweatherstation.php?" + requestQuery)
+```
+
+### Rule examples
```

Comment:
The readme should also contain a "Full Example" section for textual configuration. This should contain examples for the .things and .items file.

Reply:
Please provide one. I have never had a use for it, so have no idea. Configuration is now fully automatic and the GUI is adequate.

Reply:
There are many people who are using textual configuration. A binding must be configureable via text files. So, testing the own binding with textual configuration and use it as an example is a good proof the binding fulfils this requirement.

Reply:
I gave it a try with a .things file, and it would look like this:

`Thing wundergroundupdatereceiver:wundergroundUpdateReceiver:W1 "My weather station W1" [ stationId="W1" ]`.

But it doesn't work properly. It is possible to configure a thing via .thing file, but the channels are not recognized at the end.

.thing file generated thing:
![image](https://user-images.githubusercontent.com/48569432/133330813-ffdf8881-6ae5-4b21-a1da-464620ad34c5.png)

But no channels:
![image](https://user-images.githubusercontent.com/48569432/133331348-a00483ad-650c-44cb-868a-f1c6980707f9.png)

Auto discovery generated thing:
![image](https://user-images.githubusercontent.com/48569432/133331861-df4418bc-d396-411d-b75b-add37e39cbb9.png)

With all the channels, working like a charm ;-) :
![image](https://user-images.githubusercontent.com/48569432/133332077-a0a3aa6e-562a-4290-ac1b-5716bbb653bc.png)

Reply:
So maybe one sould document that using a .thing file is not recommended for this binding?

Reply:
No, let's please fix the underlying problem :-)

IRC it is required that a binding support textual configuration.

Reply:
And I'm sure it can be done, as long as you manually replicate the configuration discovery produces. Otherwise the binding is more or less useless, as you will not receive any values. Why would you want to do that manually? What does a thing file provide that discovery and the UI doesn't, apart from moving to other instances? If it were possible to export a live thing configuration to a file, it might be useful, but I understand that is neither supported or planned.
I would draw your attention to https://community.openhab.org/t/import-export-things-items-rules-at-openhab3/109175/14, where it seems ambiguous what the preferred mode of configuration is, but in the case at hand manual configuration is both laborious, error-prone and depends on which urlparameters the device submits, so any description is going to be complicated and .

Reply:
I think the discussion about the benefits of UI configuration and textual configuration should be kept in the forum. If a binding shall become part of the official distribution, it's mandatory to work with UI configuration and textual configuration. There are many users who use textual configuration.

Reply:
I have all my Openhab configuration organized in files. The use of files has the advantage that on can start with a fresh Openhab installation and then just restore the previously saved configuration, to get everything set up again as before, without doing the same settings again in the UI.

Reply:
@seimenb Discovery should yield identical results for a given targetted WU account including ids.